andela/ah-the-jedi-backend
# #165305274 Users should be able to tag their articles

### What does this PR do?
Implements tag an article functionality

### Description of Task(s) to be completed?
- ensure authors can add tags to their articles
- implement functionality to create tags during article creation
- implement functionality to add tags during article update
 - implement get all tags endpoint
 - update filter to search via the new tagList field
 - add unit tests

### How should this be manually tested?

Clone this repository on local machine and CD into project folder
`$ git clone git@github.com:andela/ah-the-jedi-backend.git`
`$ cd ah-the-jedi-backend`

Check out to the PR's branch
`$ git checkout ft-article-tagging-165305274`

Create and activate virtual environment
`$ python3 -m venv venv`
`$ source .venv/bin/activate`

Install app dependencies from requirements.txt
`$ pip install -r api/requirements.txt`

Make migrations and run the server
`$ python manage.py makemigrations`
`$ python manage.py migrate`
`$ python manage.py runserver`

**Use postman to test the endpoints**


           POST <your-localhost>/api/users/



       POST <your-localhost>/api/login/ 

       Use the token provided after login for authorization 

- [x] **create an article and include taglist in payload**

      `POST`  `<you-local-host>/api/articles/`

- [x] **udate an article and include taglist in payload**

      `PUT`  `<you-local-host>/api/articles/<slug>/`

- [x] **get all tags**

      `GET`  `<you-local-host>/api/tags/`


### What are the relevant PT stories?

[#165305274](https://www.pivotaltracker.com/story/show/165305274)

### Screenshots

#### Post

<img width="1113" alt="Screenshot 2019-05-13 at 12 24 11" src="https://user-images.githubusercontent.com/11965186/57610818-2d6ba500-757a-11e9-86ab-0de8dac7d184.png">

#### Get all tags

<img width="924" alt="Screenshot 2019-05-13 at 12 22 14" src="https://user-images.githubusercontent.com/11965186/57610921-6572e800-757a-11e9-987f-ec9f7e6bf4e4.png">
